anchore_engine/analyzers/syft/__init__.py

```diff
@@ -2,22 +2,22 @@
 
 from anchore_engine.analyzers.utils import defaultdict_to_dict, content_hints
 from anchore_engine.clients.syft_wrapper import run_syft
-from .handlers import handlers_by_artifact_type, handlers_by_engine_type
+from .handlers import modules_by_artifact_type, modules_by_engine_type
 
 
 def filter_artifacts(artifact):
-    return artifact["type"] in handlers_by_artifact_type
+    return artifact["type"] in modules_by_artifact_type
 
 
-def catalog_image(image, unpackdir):
+def catalog_image(imagedir, unpackdir):
     """
     Catalog the given image with syft, keeping only select artifacts in the returned results.
     """
-    all_results = run_syft(image)
+    all_results = run_syft(imagedir)
     return convert_syft_to_engine(all_results, unpackdir)
 
 
-def convert_syft_to_engine(all_results, unpackdir, handle_hints=True):
+def convert_syft_to_engine(all_results, unpackdir):
```

Comment:
This change breaks the image import path that uses the handle_hints=False case. Need to update analyzer/imports.py's invocation and make the content_hints() function able to execute outside of an actual analysis directory structure, or restore the flag to bypass it.

Reply:
Great catch, I didn't see it used in the analysis pass and believed it to be dead code. I'll put that back.

Reply:
re: https://github.com/anchore/anchore-engine/pull/811#discussion_r546637218 , I'll change `convert_syft_to_engine` to not contain the hints processing